src/components/TodoItem.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { useTodos } from '../contexts/TodoContext';
+import { deleteTodo, updateTitleTodo } from '../api/todos';
+import { ErrorMessages } from '../types/ErrorMessages';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    tempTodo,
+    removeTodo,
+    loadingTodosIds,
+    changeCompleteTodo,
+    setTodos,
+    showError,
+    setIsLoading,
+    setLoadingTodosIds,
+  } = useTodos();
+  const { id, title, completed } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedText, setEditedText] = useState(title);
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedText(event.target.value);
+  };
+
+  const updateTitle = () => {
+    setIsLoading(true);
+    setLoadingTodosIds(prev => [...prev, id]);
+    setEditedText(title);
+
+    if (!editedText.trim()) {
+      deleteTodo(id)
+        .then(() =>
+          setTodos(prevTodos =>
+            prevTodos.filter((todoItem: Todo) => todoItem.id !== id),
+          ),
+        )
+        .catch(() => {
+          showError(ErrorMessages.DeleteTodo);
+          setIsEditing(true);
+        })
+        .finally(() => {
+          setLoadingTodosIds(prevTodosIds =>
+            prevTodosIds.filter(todoId => todoId !== id),
+          );
+        });
+
+      return;
+    }
+
+    updateTitleTodo(id, editedText)
+      .then(response => setTodos(prevTodos => [...prevTodos, response]))
+      .catch(() => {
+        showError(ErrorMessages.UpdateTodo);
+        setIsEditing(true);
+      })
+      .finally(() => {
+        setLoadingTodosIds(prev => prev.filter(todoId => todoId !== id));
+      });
+  };
+
+  const handleEscape = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setIsEditing(false);
+
+      return;
+    }
+  };
+
+  if (tempTodo) {
+    loadingTodosIds.push(tempTodo.id);
+  }
```

Comment:
I think this code can potentially be a problem in some cases. 

Can we rewrite it?